### ISSUE TITLE
Improve performance by running canvas.renderAll only once per frame

### ISSUE THREAD

**Issue by mnmtanish** [kangax/fabric.js#3945]
Hi,

I'm checking out fabricjs these days for a project we're planning to do. While experimenting with Fabric and Chrome DevTools, I came across following.

![screen shot 2017-05-18 at 4 45 35 pm](https://cloud.githubusercontent.com/assets/136691/26206381/4d37abee-3c02-11e7-926b-4b8fd825325f.png)

It appears that canvas.renderAll is running multiple times per browser render frame. It's triggered by the mouse event. I tried adding the following hack and the frame rate increased a bit.

```js
let isRendering = false;
const render = canvas.renderAll.bind(canvas);

canvas.renderAll = () => {
    if (!isRendering) {
        isRendering = true;
        requestAnimationFrame(() => {
            render();
            isRendering = false;
        });
    }
};
```

Is it possible to improve FabricJS performance when using mouse events this way? Thanks.

**Comment by asturur** [comment]
yes we have already a ticket for this that is about implememting requestAnimationFrame

**Comment by asturur** [comment]
duplicate of #1979

**Comment by asturur** [comment]
is something i have to do anyway soon
